src/optimization.hpp

```diff
@@ -5,16 +5,17 @@
 
 #include <cmath>
 #include <functional>
+#include <numeric>
 #include <utility>
 
 #include "sugar.hpp"
 
 namespace Optimization {
 
-// Copied from https://www.boost.org/doc/libs/1_73_0/boost/math/tools/minima.hpp
-template <class F, class T>
-std::pair<T, T> BrentMinimize(F f, T min, T max, int significant_digits,
-                              size_t max_iter) {
+// Adapted from https://www.boost.org/doc/libs/1_73_0/boost/math/tools/minima.hpp
```

Comment:
Did you come up with the Brent + gradient yourself? If not, is there some reference?

Reply:
I modified the algorithm myself. I think there is some version of Brent + gradients out there (replacing the secant method with Newton's method in the root-finding version of Brent), but that's not what we're doing here.